treq/testing.py

```diff
@@ -89,54 +76,54 @@ def check_already_called(r):
             host, port, factory, timeout, bindAddress = (
                 self._memoryReactor.tcpClients[-1])
 
-        # Then we need to convince that factory it's connected to something and
-        # it will give us a protocol for that connection.
-        protocol = factory.buildProtocol(None)
-
-        # We want to capture the output of that connection so we'll make an
-        # in-memory transport.
-        clientTransport = AbortableStringTransport()
-        if scheme == "https":
-            directlyProvides(clientTransport, ISSLTransport)
-
-        # When the protocol is connected to a transport, it ought to send the
-        # whole request because callers of this should not use an asynchronous
-        # bodyProducer.
-        protocol.makeConnection(clientTransport)
-
-        # Get the data from the request.
-        requestData = clientTransport.io.getvalue()
+        serverAddress = IPv4Address('TCP', '127.0.0.1', port)
+        clientAddress = IPv4Address('TCP', '127.0.0.1', 31337)
+
+        # Create the protocol and fake transport for the client and server,
+        # using the factory that was passed to the MemoryReactor for the
+        # client, and a Site around our rootResource for the server.
+        serverProtocol = Site(self._rootResource).buildProtocol(None)
+        serverTransport = iosim.FakeTransport(
+            serverProtocol, isServer=True,
+            hostAddress=serverAddress, peerAddress=clientAddress)
+        clientProtocol = factory.buildProtocol(None)
+        clientTransport = iosim.FakeTransport(
+            clientProtocol, isServer=False,
+            hostAddress=clientAddress, peerAddress=serverAddress)
+
+        # Twisted 13.2 compatibility.
```

Comment:
If this is only for 13.2, shouldn't we avoid overwriting that method if it's already set?